src/components/split-axis-events.js

```diff
@@ -1,30 +1,376 @@
+  if ((angle >= 0 && angle < 120) || angle >= 330) {
+    direction += "north";
+  }
+  if (angle >= 150 && angle < 300) {
+    direction += "south";
+  }
+  if (angle >= 60 && angle < 210) {
+    direction += "west";
+  }
+  if ((angle >= 240 && angle < 360) || angle < 30) {
+    direction += "east";
+  }
+  return direction;
+};
+
+AFRAME.registerComponent("dpad-as-axes", {
+  schema: {
+    inputName: { default: "dpad" },
+    name: { default: "dpad_axes" },
+    emitter: { default: "#left-hand" }
+  },
+
+  init: function() {
+    this.mapping = [
+      {
+        direction: "north",
+        axes: [0, 1]
+      },
+      {
+        direction: "northeast",
+        axes: [1, 1]
+      },
+      {
+        direction: "east",
+        axes: [1, 0]
+      },
+      {
+        direction: "southeast",
+        axes: [1, -1]
+      },
+      {
+        direction: "south",
+        axes: [0, -1]
+      },
+      {
+        direction: "southwest",
+        axes: [-1, -1]
+      },
+      {
+        direction: "west",
+        axes: [-1, 0]
+      },
+      {
+        direction: "northwest",
+        axes: [-1, 1]
+      }
+    ];
+    this.handlers = [];
+  },
+
+  play: function() {
+    var inputName = this.data.inputName;
+    for (var pair of this.mapping) {
+      this.handlers[pair.direction] = this.emitAxes(pair.axes).bind(this);
+      this.el.addEventListener(
+        `${inputName}_${pair.direction}`,
+        this.handlers[pair.direction]
+      );
+    }
+  },
+
+  pause: function() {
+    var inputName = this.data.inputName;
+    for (var pair of this.mapping) {
+      this.el.removeEventListener(
+        `${inputName}_${pair.direction}`,
+        this.handlers[pair.direction]
+      );
+    }
+  },
+
+  emitAxes: function(axes) {
+    const name = this.data.name;
+    const inputName = this.data.inputName;
+    const emitter = document.querySelector(this.data.emitter);
+    return function(event) {
+      emitter.emit(name, { axis: [axes[0], axes[1]] });
+    };
+  }
+});
+
+AFRAME.registerComponent("wasd-dpad", {
+  schema: {
+    north: { default: "w" },
+    east: { default: "d" },
+    south: { default: "s" },
+    west: { default: "a" }
+  },
+
+  init: function() {
+    this.onKeyPress = this.onKeyPress.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.keys = {};
+  },
+
+  play: function() {
+    window.addEventListener("keypress", this.onKeyPress);
+    window.addEventListener("keyup", this.onKeyUp);
+  },
+
+  pause: function() {
+    window.remove("keypress", this.onKeyPress);
+    window.remove("keyup", this.onKeyUp);
+  },
+
+  tick: function(t, dt) {
+    const { north, east, south, west } = this.data;
+    var direction = "";
+    direction += this.keys[north] ? "north" : "";
+    direction += this.keys[south] ? "south" : "";
+    direction += this.keys[east] ? "east" : "";
+    direction += this.keys[west] ? "west" : "";
+    if (direction !== "") {
+      this.el.emit(`dpad_${direction}`);
+    }
```

Comment:
I think it might be okay to `emit` on every tick, though arguably you can throttle this a considerably. Key repetition events are typically on the order of 400ms. 
Even still, I would try to avoid all this string manipulation and only re-construct strings when the keys actually change. Worth profiling first though.